staging/src/k8s.io/component-base/metrics/testutil/metrics.go

```diff
@@ -192,23 +196,34 @@ func GetHistogramFromGatherer(gatherer metrics.Gatherer, metricName string) (His
 	}
 
 	if metricFamily == nil {
-		return Histogram{}, fmt.Errorf("metric %q not found", metricName)
-	}
-
-	if metricFamily.GetMetric() == nil {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
+		return histMap, fmt.Errorf("metric %q not found", metricName)
 	}
 
 	if len(metricFamily.GetMetric()) == 0 {
-		return Histogram{}, fmt.Errorf("metric %q is empty", metricName)
-	}
-
-	return Histogram{
+		return histMap, fmt.Errorf("metric %q is empty", metricName)
+	}
+
+	metrics := metricFamily.GetMetric()
+	if metricName == frameworkExtensionPointDurationMetricName {
```

Comment:
lets not hardcode metric names here, this file is in a generic metrics component. You could change the function signature to pass the label. If no label is provided, return the metric at index zero (i.e., the old code path)

Reply:
+1 on not hardcoding metric names. An alternative way would be to modify this function to take a metric name and a map of labels to label values, since it looks like you are selecting the histogram which matches a given label-labelValue pair.

Reply:
Done, thanks both for your suggestion.